dotCMS/src/main/java/com/dotcms/cache/VanityUrlCache.java

```diff
@@ -0,0 +1,62 @@
+package com.dotcms.cache;
+
+import com.dotcms.vanityurl.model.CachedVanityUrl;
+import com.dotcms.vanityurl.model.VanityUrl;
+import com.dotmarketing.business.Cachable;
+import java.util.Set;
+
+/**
+ * This cache is used to map the Vanity URLs path to the Vanity Url
+ * content.
+ *
+ * @author oswaldogallango
+ */
+public abstract class VanityUrlCache implements Cachable {
+
+    /**
+     * Add or update in the cache the given Vanity URL
+     * based on given the given key
+     *
+     * @return CachedVanityUrl object
+     */
+    abstract public CachedVanityUrl add(String key, VanityUrl vanityUrl);
+
+    /**
+     * Retrieves the Cached Vanity URL associated to the given
+     * key
+     *
+     * @return CachedVanityUrl
+     */
+    abstract public CachedVanityUrl get(String key);
+
+    /**
+     * Removes all entries from cache
+     */
+    abstract public void clearCache();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)